src/PerformanceSensitiveAnalyzers/Core/AbstractAllocationAnalyzer.cs

```diff
@@ -152,4 +150,65 @@ private static PerformanceSensitiveInfo CreatePerformanceSensitiveInfo(Attribute
             }
         }
     }
+
+    internal abstract class AbstractAllocationAnalyzer<TLanguageKindEnum>
```

Comment:
💡  This should be in its own file

Reply:
The end goal is to completely remove the AbstractAllocationAnalyzer<TLanguageKindEnum>. It cannot be done before all analyzers are converted though. GenericAbstractAllocationAnalyzer.cs?AbstractAllocationAnalyzer.Generic.cs?

Reply:
I suggest the file name match the actual type name: **AbstractAllocationAnalyzer`1.cs**.